looker/resource_looker_user_attribute.go

```diff
@@ -0,0 +1,241 @@
+package looker
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/go-cty/cty"
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	sdk "github.com/looker-open-source/sdk-codegen/go/sdk/v4"
+	"github.com/pkg/errors"
+	"github.com/resolutionlife/terraform-provider-looker/internal/conv"
+)
+
+func resourceUserAttribute() *schema.Resource {
+	return &schema.Resource{
+		Description:   "Manages Looker User Attributes",
+		CreateContext: resourceUserAttributeCreate,
+		ReadContext:   resourceUserAttributeRead,
+		UpdateContext: resourceUserAttributeUpdate,
+		DeleteContext: resourceUserAttributeDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The unique id of the user attribute",
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The name of the user attribute, this is how you reference the attribute in Looker expressions and LookML",
+			},
+			"label": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The user-friendly name displayed in the app for lists and filters",
+			},
+			"data_type": {
+				Type:             schema.TypeString,
+				Required:         true,
+				Description:      "Type of user attribute ('string', 'number', 'datetime', 'yesno', 'zipcode')",
+				ValidateDiagFunc: validationFunc([]string{"string", "number", "datetime", "yesno", "zipcode"}),
+			},
+			"hidden": {
+				Type:        schema.TypeBool,
+				Required:    true,
+				Description: "If set the value will be treated like a password, and once set, no one will be able to decrypt and view it",
+			},
+			"user_access": {
+				Type:             schema.TypeString,
+				Required:         true,
+				Description:      "If 'None', non-admins will not be able to see the value of this attribute for themselves. If 'View' is required to use this attribute in query filters. If 'Edit', the user will be able to set their own value of this attribute, so the user attribute will not be able to be used as an access filter.",
+				ValidateDiagFunc: validationFunc([]string{"None", "View", "Edit"}),
+			},
+			"default_value": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Value when no other value is set for the user or for one of the user's groups",
+			},
+			"domain_whitelist": {
```

Comment:
As this field cannot be set to `""`, could have a validation on this field to prevent this from being set?

Reply:
Also, this field can have `RequiredWith: "hidden"` :)

Reply:
What I could be is force recreate the resource when `domain_whitelist` changes so it can always be edited?

Reply:
This makes loads of sense! As a reference, we tested what happens to resources that have dependencies on the `looker_user_attribute` and they get updated if a `looker_user_attribute` is replaced 🎉 .